src/main/java/org/prebid/server/auction/AuctionRequestFactory.java

```diff
@@ -78,9 +82,12 @@ BidRequest fillImplicitParameters(BidRequest bidRequest, RoutingContext context)
         final BidRequest result;
 
         final HttpServerRequest request = context.request();
+        final List<Imp> imps = bidRequest.getImp();
 
         final Device populatedDevice = populateDevice(bidRequest.getDevice(), request);
         final Site populatedSite = bidRequest.getApp() == null ? populateSite(bidRequest.getSite(), request) : null;
+        final List<Imp> populatedImps = populateImps(imps, request);
+        final boolean populateImps = CollectionUtils.isNotEmpty(populatedImps);
```

Comment:
We can avoid of this variable. Method `populateImps(...)` can return NULL like `populateDevice(...)`.
So, we don't need to create unnecessary empty collection and just compare `populatedImps == null` in condition below.